test/users.e2e-spec.ts

```diff
@@ -0,0 +1,230 @@
+                app,
+            );
+            await request(app.getHttpServer())
+                .get("/users")
+                .set("Cookie", [access_token, refresh_token])
+                .expect(200)
+                .expect("Content-Type", /json/)
+                .expect((res) => {
+                    expect(res.body).toBeArray;
+                });
+        });
+        it("should return 401 when user is not logged in", async () => {
+            await request(app.getHttpServer())
+                .get("/users")
+                .set("Authorization", `Bearer ${undefined}`)
+                .expect(401)
+                .expect("Content-Type", /json/);
+        });
+        it("should return 403 when non-admin user tries to access it", async () => {
+            const { access_token, refresh_token } = await loginAndGetTokens(
+                "dan@random.com",
+                "password",
+                app,
+            );
+            await request(app.getHttpServer())
+                .get("/users")
+                .set("Cookie", [access_token, refresh_token])
+                .expect(403)
+                .expect("Content-Type", /json/);
+        });
+    });
+    describe("GET /users/me - get logged in users own details", () => {
+        it("should return 200 and array of users", async () => {
+            const { access_token, refresh_token } = await loginAndGetTokens(
+                "jessica.williamson@gmail.com",
+                "password",
+                app,
+            );
+            await request(app.getHttpServer())
+                .get("/users/me")
+                .set("Cookie", [access_token, refresh_token])
+                .expect(200)
+                .expect("Content-Type", /json/)
+                .expect((res) => {
+                    expect(res.body).toBeObject;
+                });
+        });
+        it("should return 401 when user is not logged in", async () => {
+            await request(app.getHttpServer())
+                .get("/users/me")
+                .set("Authorization", `Bearer ${undefined}`)
+                .expect(401)
+                .expect("Content-Type", /json/);
+        });
+    });
+    describe("GET /users/:userId - [ role - Admin ] - gets a user with full deatils given a user id", () => {
+        it("should return 200 and a object containing user details", async () => {
+            const { access_token, refresh_token } = await loginAndGetTokens(
+                "jessica.williamson@gmail.com",
+                "password",
+                app,
+            );
+            userId = await getUseridFromEmail("leo.rowe@outlook.com");
+            await request(app.getHttpServer())
+                .get(`/users/${userId}`)
+                .set("Cookie", [access_token, refresh_token])
+                .expect(200)
+                .expect("Content-Type", /json/)
+                .expect((res) => {
+                    expect(res.body).toBeObject;
```

Comment:
should we add an expect res.body contain the correct user detail?
e.g. res.body.email.toequal("leo.rowe@outlook.com"), just to check that the endpoint is returning the correct object